packages/x-tree-view/src/internals/plugins/useTreeViewLabel/useTreeViewLabel.types.ts

```diff
@@ -0,0 +1,74 @@
+import { TreeViewPluginSignature } from '../../models';
+import { TreeViewItemId } from '../../../models';
+import { UseTreeViewItemsSignature } from '../useTreeViewItems';
+
+export interface UseTreeViewLabelPublicAPI {
+  /**
+   * Used to update the label of an item.
+   * @param {TreeViewItemId} itemId The id of the item that is being edited.
```

Comment:
I would not talk about edition here since you can use the method without using the editing UI.

```suggestion
   * @param {TreeViewItemId} itemId The id of the item to update the label of.
```